### DISCUSSION TITLE
Custom 404 Error Page

### DISCUSSION THREAD

**Discussion by xer0xde** [coollabsio/coolify#4594] [Q&A] [answered]
Hey, I hope you’re doing well! I have a quick question regarding Coolify: Is it possible to create a custom 404 page within the platform?

Thank you in advance for your help!

Best regards
xerox

**Comment by hestiacn** [comment]
Within the platform? Shouldn't 404 pages be related to your program? It's part of the area that you need to customize! What does it have to do with Coolify?

**Reply by xer0xde** [reply]
> Within the platform? Shouldn't 404 pages be related to your program? It's part of the area that you need to customize! What does it have to do with Coolify?

![image](https://github.com/user-attachments/assets/661f1938-f51b-4977-aacb-6bd2919c176e)

I think this Text is related to Coolify.

**Reply by hestiacn** [reply]
Where does it appear? Port 8000?

**Reply by xer0xde** [reply]
> Where does it appear? Port 8000?

On every subdomain

**Reply by hestiacn** [reply, marked as answer]
Then create 403 404 410 500 files in each of your domain (VM) directories, which has nothing to do with Coolify itself, just specify the file path of the relevant error in nginx or apache!

**Reply by hestiacn** [reply]
If you're new to servers, we recommend using Hestia, which makes it relatively easy to configure virtual machines and allows you to focus on writing the development code. Leave it to you for the virtual machine configuration!
Click here to enter [hestia](https://hestiacp.com)

If you want to know more about page customization, you can refer to the Chinese page I made for Hestia [please refer to](https://debian.hestiamb.org)


[custom hestia page](https://hestiamb.org/docs/introduction/defaultpageen.html)

**Reply by xer0xde** [reply]
> Then create 403 404 410 500 files in each of your domain (VM) directories, which has nothing to do with Coolify itself, just specify the file path of the relevant error in nginx or apache!

Thanks that worked

**Reply by hestiacn** [reply]
Hestia is a good server management program, so we recommend using it or using a new, free server that doesn't have any third-party programs installed. Because hestia needs to be installed on a clean server

It will automatically create the relevant error file configuration, and you just need to deploy your code program in the /home/www/web/mydomain.com/public_html/ of the virtual machine
